devtools/materialize_asset.py

```diff
@@ -0,0 +1,52 @@
+#! /usr/bin/env python
+"""Materialize one asset & its upstream deps in-process so you can debug."""
+
+import argparse
+import importlib
+
+from dagster import AssetSelection, Definitions, define_asset_job
+
+from pudl import etl
+from pudl.settings import EtlSettings
+
+
+def _parse():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("asset_id")
+    return parser.parse_args()
+
+
+def main(asset_id):
+    """Entry point.
+
+    Defines dagster context like in etl/__init__.py - needs to be kept in sync.
+
+    Then creates a job with asset selection.
+    """
+    # TODO (daz/zach): maybe there's a way to do this directly with dagster cli?
+    defs = Definitions(
+        assets=etl.default_assets,
+        resources=etl.default_resources,
+        jobs=[
+            define_asset_job(
+                name="materialize_one",
+                selection=AssetSelection.keys(asset_id).upstream(),
+                config={
+                    "resources": {
+                        "dataset_settings": {
+                            "config": EtlSettings.from_yaml(
```

Comment:
I could see a case where we might want to use this with a specific year(s) of data, so it might be nice to be able to specify a settings file

Reply:
I think we can cross this bridge when we get to it :)

Reply:
But, yes, agreed that we will get to this bridge at some point.